src/pybind/mgr/prometheus/module.py

```diff
@@ -632,6 +632,11 @@ def get_metadata_and_osd_status(self):
             if osd_dev_node and osd_dev_node == "unknown":
                 osd_dev_node = None
 
+            # fetch the devices and ids (vendor, model, serial) from the
+            # osd_metadata
+            osd_devs = osd_metadata.get('devices', '') or 'N/A'
+            osd_dev_ids = osd_metadata.get('device_ids', '') or 'N/A'
+
```

Comment:
I think leaving the labels empty if the metadata can'be retrieved would be better here. At least it would be consistent.

The get call already has the `''` default so no need for that `or` clause.

Reply:
@jan--f I thought that if a label is empty, it's not attached to the metric (label is not set vs label set with null) - this is why I added the N/A to ensure that the label is consistent and avoids any potential confusion in the query. Is that not the case? (apologies for the delay, I've been on vacation)

Reply:
Just following up on this. The goal I have is to enable the use of the device_ids field in osd down+out alerts - to direct the admin the potential physical issues relating to a osd failure. If we leave the label as an empty string, although we pass it back to prometheus, prometheus will treat the label as unset and you won't see it in the prometheus UI. By assigning an "N/A" string, the label will always be there whether the environment is physical or virtual (libvirt) - which means any alert that references the label in it's text will be more meaningful.

My goal here is to improve the data, so we can potentially improve the alerts we generate
@trociny thoughts (Boris suggested reaching out to you since you've been more active with the prometheus code recently)

Reply:
It looks like too late to make a comment (it has been merged already).

Anyway. I am also not very sure I like 'N/A' here, but I don't have much experience on the prometheus side, so if you say that the empty labels would not work and we can't think out anything better, then I just agree with you.

But you did not address the second part of @jan--f comment: why do `osd_metadata.get('devices', '') or 'N/A'` if you can just do `osd_metadata.get('devices', 'N/A')`?

Reply:
it is actually safer to use `or` there for consistency in case the `devices` were defined in `osd_metadata` but still empty. This way, you will always get `N/A`.

Reply:
I see. Thanks